tests/integration/docker_utils/test_docker.py

```diff
@@ -1370,6 +1371,58 @@ def test_docker_sdk_timeout_seconds(self, monkeypatch):
         sdk_client = SdkDockerClient()
         assert sdk_client.docker_client.api.timeout == 987
 
+    def test_docker_sdk_no_retries(self, monkeypatch):
+        monkeypatch.setattr(config, "DOCKER_SDK_DEFAULT_RETRIES", 0)
+        # change the env for the docker socket (such that it cannot be initialized)
+        monkeypatch.setenv("DOCKER_HOST", "tcp://non_existing_docker_client:2375/")
+        sdk_client = SdkDockerClient()
+        assert sdk_client.docker_client is None
+
+    def test_docker_sdk_retries_on_init(self, monkeypatch):
+        # increase the number of retries
+        monkeypatch.setattr(config, "DOCKER_SDK_DEFAULT_RETRIES", 10)
+        # change the env for the docker socket (such that it cannot be initialized)
+        monkeypatch.setenv("DOCKER_HOST", "tcp://non_existing_docker_client:2375/")
+        global sdk_client
+
+        def on_demand_init(*args):
+            global sdk_client
+            sdk_client = SdkDockerClient()
+            assert sdk_client.docker_client is not None
+
+        # start initializing the client in another thread (with 10 retries)
+        init_thread = FuncThread(func=on_demand_init)
+        init_thread.start()
+        # reset / fix the DOCKER_HOST config
+        # monkeypatch.delenv("DOCKER_HOST")
```

Comment:
I think this should be uncommented

Reply:
Thanks, of course! That's a leftover from verifying the test case.